### DISCUSSION TITLE
Does Directus support Cloudflare R2 for storage?

### DISCUSSION THREAD

**Discussion by MrJohnWi** [directus/directus#18857] [Feature Requests]
### Summary

If Directus does not currently support it, would like to know if there is a way to manually add Cloudflare R2 in Directus.

### Motivation

New storage adapter 

### Detailed Design

.......

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

........

### Alternatives

.........

### Adoption Strategy

......

**Comment by martinsotirov** [comment]
Why did you close the discussion? I'm interested in that too. Also in Cloudflare Images as a storage service for images specifically. R2 for the rest.

**Reply by 2820402** [reply]
https://discord.com/channels/725371605378924594/1117117731045658812

**Reply by martinsotirov** [reply]
It's empty on my end. And it shows `You do not have permission to send messages in this channel.` in the chat input field.

**Reply by 2820402** [reply]
Just use it like if it was S3, it just works
Yes, it's S3 compatible
This is what I use in all my Directus projects:

```bash
STORAGE_LOCATIONS=s3
STORAGE_S3_DRIVER=s3
STORAGE_S3_KEY=key
STORAGE_S3_SECRET=secret
STORAGE_S3_ENDPOINT=https://[USER-ID].r2.cloudflarestorage.com
STORAGE_S3_BUCKET=my-bucket
STORAGE_S3_REGION=weur
STORAGE_S3_ROOT=assets
```

**Reply by martinsotirov** [reply]
@2820402 thanks a lot!
